File: Prims Algorithm/C/prims.c
#include <stdio.h>
#include <limits.h>

#define V 5

int minKey(int key[], int mstSet[]) {
    int min = INT_MAX, min_index;
    for(int v = 0; v < V; v++)
        if(mstSet[v] == 0 && key[v] < min)
            min = key[v], min_index = v;
    return min_index;
}

int main() {
    int graph[V][V] = {
        {0, 2, 0, 6, 0},
        {2, 0, 3, 8, 5},
        {0, 3, 0, 0, 7},
        {6, 8, 0, 0, 9},
        {0, 5, 7, 9, 0}
    };
    int parent[V], key[V], mstSet[V];
    for(int i = 0; i < V; i++) key[i] = INT_MAX, mstSet[i] = 0;
    key[0] = 0;
    parent[0] = -1;

    for(int count = 0; count < V - 1; count++) {
        int u = minKey(key, mstSet);
        mstSet[u] = 1;
        for(int v = 0; v < V; v++)
            if(graph[u][v] && mstSet[v] == 0 && graph[u][v] < key[v])
                parent[v] = u, key[v] = graph[u][v];
    }

    for(int i = 1; i < V; i++)
        printf("%d - %d\n", parent[i], i);

    return 0;
}

File: Prims Algorithm/Python/prims.py
V = 5

def minKey(key, mstSet):
    min_val = float('inf')
    min_index = -1
    for v in range(V):
        if key[v] < min_val and not mstSet[v]:
            min_val = key[v]
            min_index = v
    return min_index

graph = [
    [0, 2, 0, 6, 0],
    [2, 0, 3, 8, 5],
    [0, 3, 0, 0, 7],
    [6, 8, 0, 0, 9],
    [0, 5, 7, 9, 0]
]

key = [float('inf')] * V
parent = [-1] * V
mstSet = [False] * V
key[0] = 0

for _ in range(V - 1):
    u = minKey(key, mstSet)
    mstSet[u] = True
    for v in range(V):
        if graph[u][v] and not mstSet[v] and graph[u][v] < key[v]:
            key[v] = graph[u][v]
            parent[v] = u

for i in range(1, V):
    print(parent[i], "-", i)
